src/scripts/import-sdr-shipments-sheet/add-countries.ts

```diff
@@ -33,40 +35,32 @@ export async function addCountries(shipments: ShipmentCsv[]) {
   );
 
   // { "SUCCESS": [], "ALREADY_EXISTS": [], ...}
-  const resultsMap: CountryUploadWorkflowResults = Object.keys(
-    UploadWorkflowStatus,
-  ).reduce((resultsMap, key) => {
-    resultsMap[key] = [];
-    return resultsMap;
-  }, {} as CountryUploadWorkflowResults);
-
-  results
-    .map((result) => {
-      if (isFulfilled(result)) {
-        return result.value;
-      } else {
-        return result.reason;
-      }
-    })
-    .reduce((resultsMap, workflowResult) => {
-      if (workflowResult.status) {
-        resultsMap[workflowResult.status].push(workflowResult);
-      } else {
-        resultsMap[UploadWorkflowStatus.OTHER].push(workflowResult);
-      }
-      return resultsMap;
-    }, resultsMap);
+  const resultsMap: CountryUploadWorkflowResults = {
+    ...(_.chain(UploadWorkflowStatus)
+      .keys()
+      .map((key) => [key, [] as Country[]])
+      .fromPairs()
+      .value() as CountryUploadWorkflowResults),
+    ..._.chain(results)
+      .map((res) => (isFulfilled(res) ? res.value : res.reason))
+      .groupBy((res) => res.status ?? UploadWorkflowStatus.OTHER)
+      .value(),
+  };
```

Comment:
@corasaurus-hex love the simplicity of this functional style workflow.  However I'm nervous about such terse code not being accessible to less experienced devs (or even just me trying to remember what's going on here 3 months from now).  For example, the expansion syntax with lodash calls (`..._.chain`, etc) or why we're expanding 2 sets of values into the object (to ensure every key has an entry, and then to override entries where there are actual results with that status).

**question:** Could we break up the `_.chain` functions into variable assignments to provide clarity here?  Or add in some comments or something?  Open to ideas!

Reply:
Totally fair, I just reverted it entirely

Reply:
@corasaurus-hex that seems like a good approach for the moment to get this merged in.

I do really like your suggestion of using Lodash.  Maybe we try to introduce it overall and develop some conventions, then re-approach refactoring this script to use it if needed?